File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)


{
    float n;
    int i;
    
    do 
    {
        printf("O hai! How much change is owed?\n");
        n = get_float();
    }
    while ( n < 0 ); 
    
    n = roundf(n * 100);
    i = 0;
    
    
     while (n >= 25) {
        n = n - 25;
        i = i + 1;
        
    };
     while (n >= 10) {
        n = n - 10;
        i = i + 1;
        
    };
     while (n >= 5) {
        n = n - 5;
        i = i + 1;
        
    };
    while (n >= 1) {
        n = n - 1;
        i = i + 1;
        
    };
    
    
    printf("%i\n", i);
   
}

File: pset6/greedy.py
import cs50

while True:
    change = float(input("O hai! How much chnage is owed? "))
    if change > 0:
        break

change = round(change * 100)
coins = 0

#print(change);

while change > 0:
    while change >= 25:
        change = change - 25
        coins = coins + 1
        
    while change >= 10:
        change = change - 10
        coins = coins + 1
        
    while change >= 5:
        change = change - 5
        coins = coins + 1
        
    while change >= 1:
        change = change - 1
        coins = coins + 1
        
print(coins)
    
